### PR TITLE
Add support for subPath in secrets volumes

### PR DESCRIPTION
**Pull request by autero1**
## Description

Add support for subPath in secrets volumes, same way as already possible for configMaps.

## TODOs

Read the [Gruntwork contribution guidelines](https://gruntwork.notion.site/Gruntwork-Coding-Methodology-02fdcd6e4b004e818553684760bf691e).

- [ ] Update the docs.
- [ ] Run the relevant tests successfully, including pre-commit checks.
- [ ] Ensure any 3rd party code adheres with our [license policy](https://www.notion.so/gruntwork/Gruntwork-licenses-and-open-source-usage-policy-f7dece1f780341c7b69c1763f22b1378) or delete this line if its not applicable.
- [ ] Include release notes. If this PR is backward incompatible, include a migration guide.

## Release Notes (draft)

Added / Removed / Updated [X].

### CODE REVIEW

**Comment by autero1** [comment]
All the tests passed:
image

File: test/k8s_service_config_injection_template_test.go [hunk 1]
```diff
@@ -194,6 +195,44 @@ func TestK8SServiceVolumeConfigMapAddsVolumeAndVolumeMountWithoutSubPathToPod(t
 	assert.Empty(t, volumeMount.SubPath)
 }
 
+// Test that setting the `secrets` input value with volume include the volume mount for the secret
```
Comment (zackproser) [inline test/k8s_service_config_injection_template_test.go:198]:
```suggestion
// Test that setting the `secrets` input value with volume includes the volume mount for the secret
```

File: test/k8s_service_config_injection_template_test.go [hunk 2]
```diff
@@ -234,6 +273,46 @@ func TestK8SServiceVolumeConfigMapAddsVolumeAndVolumeMountWithSubPathToPod(t *te
 	assert.Equal(t, volumeMount.SubPath, "host.txt")
 }
 
+// Test that setting the `secrets` input value with volume include the volume mount and subpath for the secret
```
Comment (zackproser) [inline test/k8s_service_config_injection_template_test.go:276]:
```suggestion
// Test that setting the `secrets` input value with volume includes the volume mount and subpath for the secret
```

**Review by zackproser — APPROVED** [review]
LGTM

**Comment by autero1** [comment]
Thanks for the review! Merging this in.

**Pull request merged**